### PR TITLE
[torchlib] Add missing operators (set 2)

### PR DESCRIPTION
**Pull request by shubhambhokare1**
- [x] aten.special.expm1
- [x] aten.sort

### CODE REVIEW

File: onnxscript/function_libs/torch_lib/ops/core.py [hunk 1]
```diff
@@ -7699,12 +7699,23 @@ def aten_softmax_no_dtype(self: TFloatOrBFloat16, dim: int) -> TFloatOrBFloat16:
...
+) -> tuple[TReal, INT64]:
+    """sort(Tensor self, int dim=-1, bool descending=False, bool stable=False) -> (Tensor values, Tensor indices)"""
 
-    raise NotImplementedError()
+    self_is_scalar = IsScalar(self)
+    if self_is_scalar:
+        self = op.Unsqueeze(self, op.Constant(value_ints=[0]))
+    shape = op.Shape(self)
+    dim_size = op.Gather(shape, dim, axis=0)
+    dim_size = op.Reshape(op.Cast(dim_size, to=INT64.dtype), op.Constant(value_ints=[1]))
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_lib/ops/core.py:7713]:
```suggestion
    dim_size = op.Reshape(dim_size, op.Constant(value_ints=[1]))
```

op.Shape should always be int64

File: onnxscript/function_libs/torch_lib/ops/core.py [hunk 2]
```diff
@@ -7699,12 +7699,23 @@ def aten_softmax_no_dtype(self: TFloatOrBFloat16, dim: int) -> TFloatOrBFloat16:
...
 def aten_sort(
-    self: TensorType, dim: int = -1, descending: bool = False
-) -> tuple[TensorType, TensorType]:
-    """sort(Tensor self, int dim=-1, bool descending=False) -> (Tensor values, Tensor indices)"""
+    self: TReal, dim: int = -1, descending: bool = False, stable: bool = False
+) -> tuple[TReal, INT64]:
+    """sort(Tensor self, int dim=-1, bool descending=False, bool stable=False) -> (Tensor values, Tensor indices)"""
 
-    raise NotImplementedError()
+    self_is_scalar = IsScalar(self)
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_lib/ops/core.py:7708]: if self is scalar, is it possible to just return self? e.g. op.Identity(self) ?

**Review by justinchuby — APPROVED** [review]

File: onnxscript/function_libs/torch_lib/ops/core.py (hunk 2 above)
Comment (shubhambhokare1) [reply onnxscript/function_libs/torch_lib/ops/core.py:7708]: Had tried that initially, but looks like we cannot return from inside a control-flow block

Comment (gramalingam) [reply onnxscript/function_libs/torch_lib/ops/core.py:7708]: Should it be made traceonly? The only control-flow logic is rank-dependent, and we are anyway assuming that is known as export-time. Then, you can do what Justin suggests.

Comment (justinchuby) [reply onnxscript/function_libs/torch_lib/ops/core.py:7708]: Alternatively, you can assign the output of identity to result, then return result outside of the control flow. But trace only makes sense

File: onnxscript/function_libs/torch_lib/ops/core.py [hunk 3]
```diff
@@ -7699,12 +7699,20 @@ def aten_softmax_no_dtype(self: TFloatOrBFloat16, dim: int) -> TFloatOrBFloat16:
...
+    if self_is_scalar:
+        return op.Identity(self), op.Squeeze(op.Constant(value_ints=[0]))
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_lib/ops/core.py:7710]:
```suggestion
        return op.Identity(self), op.Constant(value_int=0)
```

File: onnxscript/function_libs/torch_lib/ops/core.py [hunk 4]
```diff
@@ -7699,12 +7699,20 @@ def aten_softmax_no_dtype(self: TFloatOrBFloat16, dim: int) -> TFloatOrBFloat16:
...
+) -> tuple[TReal, INT64]:
+    """sort(Tensor self, int dim=-1, bool descending=False, bool stable=False) -> (Tensor values, Tensor indices)"""
 
-    raise NotImplementedError()
+    self_is_scalar = IsScalar(self)
+    if self_is_scalar:
+        return op.Identity(self), op.Constant(value_int=0)
+    shape = op.Shape(self)
+    dim_size = op.Gather(shape, dim, axis=0)
+    dim_size = op.Reshape(dim_size, op.Constant(value_ints=[1]))
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_lib/ops/core.py:7713]: Use a slice to replace the two ops?

Comment (shubhambhokare1) [reply onnxscript/function_libs/torch_lib/ops/core.py:7713]: Oh, think I missed this comment before merge. Making a note of this as a #TODO for the PR covering the next set of operators.

**Pull request merged**